src/content/1.1/code/ocaml/snippet04.ml

```diff
@@ -0,0 +1,13 @@
+module Compose_Three_GF = functor(F:Polymorphic_Function_F)(G:Polymorphic_Function_G with type b = F.b)(H:Polymorphic_Function_H with type c = G.c) -> struct
+  let (>>) : 'a 'b 'c. ('b -> 'c) -> ('a -> 'b) -> 'a -> 'c = 
+    fun g f x -> g (f x)
+  let compose : 'a -> 'd = H.h >> (G.g >> F.f)
+end
+
+module Compose_Three_HG = functor(F:Polymorphic_Function_F)(G:Polymorphic_Function_G with type b = F.b)(H:Polymorphic_Function_H with type c = G.c) -> struct
+  let (>>) : 'a 'b 'c. ('b -> 'c) -> ('a -> 'b) -> 'a -> 'c = 
+    fun g f x -> g (f x)
+  let compose : 'a -> 'd = (H.h >> G.g) >> F.f
+end
+
+Compose_Three_GF.compose = Compose_Three_HG.compose
```

Comment:
Do we really need all this for `compose`? :)

Reply:
OCaml doesn't have a compose operator. I added a comment and removed some redundant compose declarations.

Reply:
Hi!
Just started reading the book. OCaml does have a composition operator, its `@@`. It was added in 4.01, released in september 2013.

> Application operator: `g @@ f @@ x` is exactly equivalent to `g (f (x))`.

https://caml.inria.fr/pub/docs/manual-ocaml/libref/Stdlib.html#VAL(@@)

Reply:
But `let compose = g @@ f` is not equivalent. You , need `let compose = fun x -> f @@ g x`, so for all the compositionality examples I think it would be better to keep a `(>>)` in a hypothetical prelude

Reply:
I see what you mean. You’re right.

Reply:
Even then you still run into issues with the value restriction. There's a very (very) long discussion about composition in OCaml [here](https://discuss.ocaml.org/t/why-dont-we-have-a-composition-operator-in-pervasives/1210) (post discussing value restriction [here](https://discuss.ocaml.org/t/why-dont-we-have-a-composition-operator-in-pervasives/1210/35)).